examples/prompt_non_blocking.cpp

```diff
@@ -0,0 +1,41 @@
+#include <cpp-terminal/version.h>
+#include <cpp-terminal/prompt.hpp>
+#include <iostream>
+
+int main() {
+    std::cout << "Running cpp-terminal version: "
+              << CPP_TERMINAL_VERSION_COMPLETE << std::endl;
+    try {
+        std::cout << "CPP-Terminal basic prompt example: \n\n";
+        switch (Term::prompt_non_blocking("Proceed?", "Y", "n", ":")) {
```

Comment:
is this blocking in the sense of "blocking call" or wether `Enter` has to be pressed or not?

If the latter, I think another name might be better. 

It might be good to have a "non-blocking" prompt, too, so that one could cancel after some timeout or similar.

Reply:
I have created notes in the `cpp-terminal/prompt.hpp` file on that. Basically "blocking" is that the the function is waiting for the user to hit the enter key - like i a normal console. The non-blocking version is immediately returning after the first key is pushed.

Maybe we should rename the function into something better describing...
I'd suggest to add another argument for adding a timeout for the function.

Reply:
I think if I read "blocking" or "non-blocking" I would think that the program can continue while the prompt is shown (regardless wether Enter is required or not). So yeah, maybe we can find a better name...

Reply:
Any idea for a name? Will add a simple timeout after I fixed the windows CI.

Reply:
something like `immediate_prompt`? or `prompt_need_confirm` or `prompt_enter` or one could also make it an argument. 
I also quite like `structs` as containers for options, such as 

```
struct prompt_options {
bool needs_enter = false;
bool multiline = false;
bool simple = true; ...
}
```

Reply:
Another way is via flags, such as `NEED_ENTER | MULTILINE`, and then each of those is an integer with a bit flag. 

But you guys need to decide that :)

Reply:
Perhaps the easiest is just boolean arguments, possibly with default values.

Reply:
adding an option for the multi-line prompt isn't that of of a good i idea I would say. It's really complicated. I'll add a bool for toggling the confirmation. I think that a struct would be more complicated than just passing 5/6 short arguments.